src/service/cid-impl.js

```diff
@@ -155,6 +156,16 @@ export class Cid {
               rethrowAsync(error);
             });
       });
+    }).catch(error => {
+      const docVisible = viewer.isVisible();
+      const hasVisible = viewer.hasBeenVisible();
+      trace.message += error.message;
+      trace.message +=
+          ` EXTRA INFO: doc isVisible: ${docVisible},` +
+          ` doc hasBeenVisible ${hasVisible}`;
+      dev().error('CID', trace);
+      rethrowAsync(error);
+      return null;
```

Comment:
this will change the behaviour. Previously the final promise would have been rejected, not it will be satisfied with null. (which I guess is similar to if timeout happened, except timeout will resolve with `undefined`).

Reply:
I don't want this. But then w/o `return null` I can fix the type-check error. Seems the function expects a return of `{!Promise<?string>} `. Any suggestion on how to skip it?

Reply:
Whichever way typecheck issue is solved here, this method must throw an exception, i.e. `throw error`. `rethrowAsync` is no good. But a bigger question I have: are you sure that error arrives here at all? I see another `catch` with `rethrowAsync` in the old code above... Can you retest? It possible that we simply need to move this `dev().error()` there?

Reply:
I tested  with `this. getExternalCid_()` returning an error. I am able to see error arrives here.

Reply:
I'll see what I can do with the type check to just `throw error`. And the other `rethrowAsync` is specially related  to timeout error, so we should be good.

Reply:
Ok I catch the error earlier, and only catch the error from `this.getExternalCid_`. Which makes thing more clear, fix type-check issue. 
Tested locally and able to see the error.

Reply:
I still a bit doubtful about this. How do you throw an error from `getExternalCid_` for testing? You should try this via `return Project.reject('TEST')` b/c this most closely correlates with errors inside messaging pipe.

Reply:
Ahhhh I see. I tried with `throw new Error()`. 
Tried `return Project.reject()`, you are right, we can simply move `dev().error()` to the old `catch` block to avoid throwing duplicate error. 
Fixed.